lib/IDE/SyntaxModel.cpp

```diff
@@ -568,15 +571,9 @@ std::pair<bool, Expr *> ModelASTWalker::walkToExprPre(Expr *E) {
     pushStructureNode(SN, Elem);
   };
 
-  if (auto *ParentTupleExpr = dyn_cast_or_null<TupleExpr>(Parent.getAsExpr())) {
-    // the argument value is a tuple expression already, we can just extract it
-    addCallArgExpr(E, ParentTupleExpr);
-  } else if (auto *ParentOptionalExpr = dyn_cast_or_null<OptionalEvaluationExpr>(Parent.getAsExpr())) {
```

Comment:
Huh, was this check always dead? I don't really understand why it was ever needed, since presumably we would visit the `OptionalEvaluationExpr` itself...

Reply:
Yeah, I *believe* it has always been dead – it was added in https://github.com/apple/swift/pull/11542, but I'm pretty sure the real fix in that patch was the moving of the `isImplicit` check